.github/workflows/CI_on_push.yml

```diff
@@ -0,0 +1,41 @@
+
+name: CI on push to main or issue-2 branch
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - issue-2-Setup_CI/CD_for_reference-models
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v10
+
+      - name: Install clang and runtime dependencies
+        run: |
+          sudo apt install clang libunwind-dev
+          sudo apt install libgc-dev libre2-dev
+
+      - name: Checkout cli repo
+        uses: actions/checkout@v2
+        with:
+          repository: nervosum/nervosum
+          path: cli
+
+      - name: Build the CLI binary
+        run: |
+          cd cli
+          sbt cli/nativeLink
```

Comment:
Excellent approach! This way we're always using the latest version of the cli, which is exactly the purpose of this setup at this point. Perhaps we should add: `git log --oneline -1` after the `sbt` command. Than we know exactly which commit is used in the current CI pipeline.

Reply:
Nevermind, I see this is already part of the standard Github actions workflow. The logs already show which version of the repository was checked out. #learned-something-new :).